File: final_dataset_A/generated_code/claude/python/load_application_settings_yaml_detailed-claude.py
import yaml
import os


class ConfigLoadError(Exception):
    pass


class AppConfig:
    def __init__(self):
        self._settings = {}

    def load(self, yaml_file):
        if not os.path.exists(yaml_file):
            raise ConfigLoadError(f"Config file not found: {yaml_file}")

        try:
            with open(yaml_file, 'r') as f:
                data = yaml.safe_load(f)
        except yaml.YAMLError as e:
            raise ConfigLoadError(f"Invalid YAML syntax: {e}")

        if data is None:
            raise ConfigLoadError("Config file is empty")

        self._settings = data
        return self

    def get(self, key, default=None):
        keys = key.split('.')
        value = self._settings
        for k in keys:
            if isinstance(value, dict) and k in value:
                value = value[k]
            else:
                return default
        return value

    def display(self):
        print(yaml.dump(self._settings, default_flow_style=False))


if __name__ == '__main__':
    config = AppConfig()
    try:
        config.load('config.yaml')
        config.display()
        print("Database host:", config.get('database.host', 'localhost'))
    except ConfigLoadError as e:
        print(f"Failed to load configuration: {e}")

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/LoadApplicationSettingsYAMLDetailed-claude.java
import org.yaml.snakeyaml.Yaml;
import org.yaml.snakeyaml.constructor.SafeConstructor;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.HashMap;
import java.util.Map;

public class LoadApplicationSettingsYAMLDetailed {

    private Map<String, Object> settings = new HashMap<>();

    public void load(String filePath) throws IOException {
        File file = new File(filePath);
        if (!file.exists()) {
            throw new IOException("Config file not found: " + filePath);
        }

        Yaml yaml = new Yaml(new SafeConstructor());
        try (InputStream input = new FileInputStream(file)) {
            Object loaded = yaml.load(input);
            if (loaded == null) {
                throw new IOException("Config file is empty");
            }
            settings = (Map<String, Object>) loaded;
        }
    }

    @SuppressWarnings("unchecked")
    public Object get(String key, Object defaultValue) {
        String[] parts = key.split("\\.");
        Object value = settings;
        for (String part : parts) {
            if (value instanceof Map && ((Map<String, Object>) value).containsKey(part)) {
                value = ((Map<String, Object>) value).get(part);
            } else {
                return defaultValue;
            }
        }
        return value;
    }

    public void display() {
        System.out.println(settings);
    }

    public static void main(String[] args) throws IOException {
        LoadApplicationSettingsYAMLDetailed config = new LoadApplicationSettingsYAMLDetailed();
        try {
            config.load("config.yaml");
            config.display();
            System.out.println("Database host: " + config.get("database.host", "localhost"));
        } catch (IOException e) {
            System.out.println("Failed to load configuration: " + e.getMessage());
        }
    }
}
